WordPress/src/main/java/org/wordpress/android/ui/pages/PageParentFragment.kt

```diff
@@ -222,12 +226,13 @@ class PageParentFragment : Fragment() {
     private fun hideSearchList(myActionMenuItem: MenuItem) {
         recyclerView.visibility = View.VISIBLE
         frameSearch.visibility = View.GONE
-        if (saveButton?.isVisible == true) {
-            saveButton?.isVisible = false
-        }
         if (myActionMenuItem.isActionViewExpanded) {
             myActionMenuItem.collapseActionView()
         }
+        /**Force the recyclerview to redraw if selection has changed while in search mode*/
+        if (viewModel.currentParent != viewModel.initialParent) {
+            recyclerView.adapter?.notifyDataSetChanged()
```

Comment:
I wanted to propose moving this logic into the viewModel. Updating _pages with the updated list should force the view to redraw only the changed items, since the PageParentAdapter.update(..) is using DiffUtils. However, it unfortunately won't work. We currently share the list of PageItems between the VM and the adapter -> the PageItems are mutable and whenever we change their state (eg. isSelected field) in the vm it gets changed in the adapter (since they share the same instance). Therefore, the DiffUtils won't find any difference.

Having said all that, I don't think it's worth fixing at the moment as we'd need to fix it in both regular search + parent search. I just wanted to share another great example of why using mutable classes is risky and can easily cause unexpected side-effects.

Reply:
@malinajirka absolutely agree with you. DiffUtils won't work in this case.  It's a larger issue that needs to be addressed and I intentionally did not dive into (hence why you see this line for the trial). This list should be immutable. I may have even attempted a more reactive pattern to handle state changes more gracefully.